File: sharedlife-backend/src/main/java/net/tfg/sharedlife/model/Debt.java
package net.tfg.sharedlife.model;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode
@Entity
@Table(name = "debts")
public class Debt {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column(name = "idHome")
	private Long idHome;
	
	@Column(name = "idUser")
	private Long idUser; // ID DEL USUARIO QUE TIENE QUE PAGAR LA DEUDA
	
	//@Column(name = "idSpent")
	//private Long idSpent;
	
	@Column(name = "pricePerPerson")
	private double pricePerPerson;
	
	@Column(name = "paid")
	private boolean paid;
	
	@Column(name = "userToPay")
	private String userToPay;

	@ManyToOne(cascade = CascadeType.PERSIST) // Este cascade all provoca el borrado tambien del gasto asociado
	@JoinColumn(name = "spent_id")
	private Spent spent;
	
}

File: sharedlife-frontend/src/app/models/Debt/debt.ts
export class Debt {
  id: string;
  idHome: string;
  idUser: string;
  idSpent: string;
  paid: boolean;
  pricePerPerson: string;
  userToPay: string;

  constructor(i: string, ih: string, iu: string, is: string, p: boolean, ppp: string, utp: string){
    this.id = i;
    this.idHome = ih;
    this.idUser = iu;
    this.idSpent = is;
    this.paid = p;
    this.pricePerPerson = ppp;
    this.userToPay = utp;
  }

}
